tests/kernel/workq/work_timeout/src/test.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
```

Comment:
cant this be folded into some existing workq tests, adding a test per new API is a bit of an overkill, see if you can put this in one of the existing worq tests please.